### PR TITLE
[CWS] Add syscalls to activity dumps

### PR DESCRIPTION
**Pull request by Gui774ume**
### What does this PR do?

This PR adds the list of syscalls that were issued per node in an activity dump. The list of syscalls is flushed through the perf map at a configurable rate speed (see `runtime_security_config.activity_dump.syscall_monitor.period`). This feature is behind a configuration flag (see `runtime_security_config.activity_dump.syscall_monitor.enabled`).

### Motivation

Improve the feature set of activity dumps.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -1075,6 +1074,8 @@ func InitConfig(config Config) {
 	config.BindEnvAndSetDefault("runtime_security_config.activity_dump.local_storage.compression", true)
 	config.BindEnvAndSetDefault("runtime_security_config.activity_dump.remote_storage.formats", []string{"json"})
 	config.BindEnvAndSetDefault("runtime_security_config.activity_dump.remote_storage.compression", true)
+	config.BindEnvAndSetDefault("runtime_security_config.activity_dump.syscall_monitor.enabled", true)
```
Comment (vickenty) [inline pkg/config/config.go:1077]: Is it expected to be enabled by default?

Comment (Gui774ume) [reply pkg/config/config.go:1079]: For now yes, we're testing it in multiple environment and hope to be able to enable it by default on merge 👍🏻

File: pkg/security/adproto/v1/activity_dump.proto [hunk 2]
```diff
@@ -39,6 +39,7 @@ message ProcessActivityNode {
     repeated FileActivityNode files = 4;
     repeated DNSNode dns_names = 5;
     repeated SocketNode sockets = 6;
+    repeated uint32 syscalls = 7;
```
Comment (paulcacheux) [inline pkg/security/adproto/v1/activity_dump.proto:42]: How do we differentiate an `x64` syscall, with a `x86` syscall for example ?

Comment (paulcacheux) [reply pkg/security/adproto/v1/activity_dump.proto:42]: More than that, maybe we should add the arch to the metadata

Comment (Gui774ume) [reply pkg/security/adproto/v1/activity_dump.proto:42]: We don't and you're right. 😭

Comment (Gui774ume) [reply pkg/security/adproto/v1/activity_dump.proto:43]: done ✅

**Review by paulcacheux — APPROVED** [review]

**Review by djmitche — APPROVED** [review]

**Pull request merged**